public/build.gradle

```diff
@@ -18,4 +19,23 @@ dependencies {
     testImplementation 'io.mockk:mockk:1.10.0'
     testImplementation 'org.assertj:assertj-core:3.13.2'
     testImplementation project(":test-utils")
-}
\ No newline at end of file
+}
+
+tasks.dokkaHtmlPartial.configure {
```

Comment:
would be nice to pull the common configuration stuff out somehow, but i'm also not great at gradle...

Reply:
This could be extracted to a file like `library.gradle` and call it `dokka.gradle`. Then apply the file doing:

```
apply from: "$rootProject.projectDir/dokka.gradle"
```

But there's one thing that changes: the source directory. It is `src/main/java` for `public` and `src/main/kotlin` for `purchases`. I am not sure how to change that part in each of the modules. Maybe setting just that part in the module's gradle inherits everything from the `dokka.gradle` and changes that part for the module. Something like:

```
apply from: "$rootProject.projectDir/dokka.gradle"
...
dokkaSourceSets {
         configureEach {
             sourceLink {
                 localDirectory.set(file("src/main/java"))
                 remoteUrl.set(uri("https://github.com/revenuecat/purchases-android/blob/main/public/src/main/java").toURL())
             }
         }
     }
```


We should move everything to `src/main/kotlin` although not a priority at all.

As an example. I found the `arrow` project does it this way https://github.com/arrow-kt/arrow/blob/ee319ace7df8dd05001406b3727aa3ed90188e07/arrow-libs/gradle/apidoc-creation.gradle

They have a `apidoc-creation.gradle`, and they do `"            ${relativeProjectPath('src/main/kotlin')}"` to figure out the relative path of the source files. Then they apply the gradle file in each of the modules https://github.com/arrow-kt/arrow/blob/2569ece084a0ace0a77c0ed126b452babe69d52b/arrow-libs/core/arrow-annotations/build.gradle#L7. They have a gradle property https://github.com/arrow-kt/arrow/blob/main/arrow-libs/gradle.properties#L80 pointing to that file.

Reply:
agreed, I think that's the intent of the `dokkaHtmlPartial` clauses. 
I started by trying to have as much stuff in a single file as possible but kept running into issues, so I ended up going with this path for the sake of not spending too much time on this. 

I can give it another shot if you think it's important to do now, otherwise we can add it to the backlog

Reply:
I don't think it's important to do right now 🙂